lifecycle-operator/webhooks/pod_mutator/pod_mutating_webhook.go

```diff
@@ -93,16 +94,28 @@ func (a *PodMutatingWebhook) Handle(ctx context.Context, req admission.Request)
 	logger.Info(fmt.Sprintf("Pod annotations: %v", pod.Annotations))
 
 	podIsAnnotated := a.isPodAnnotated(pod)
-	logger.Info("Checked if pod is annotated.")
-
 	if !podIsAnnotated {
 		logger.Info("Pod is not annotated, check for parent annotations...")
 		podIsAnnotated = a.copyAnnotationsIfParentAnnotated(ctx, &req, pod)
 	}
 
 	if podIsAnnotated {
-		logger.Info("Resource is annotated with Keptn annotations, using Keptn scheduler")
-		pod.Spec.SchedulerName = "keptn-scheduler"
+		logger.Info("Resource is annotated with Keptn annotations")
+		if a.SchedulingGatesEnabled {
+			logger.Info("SchedulingGates enabled")
+			_, gateRemoved := getLabelOrAnnotation(&pod.ObjectMeta, apicommon.SchedulingGateRemoved, "")
+			if gateRemoved {
+				return admission.Allowed("gate of the pod already removed")
+			}
+			pod.Spec.SchedulingGates = []corev1.PodSchedulingGate{
+				{
+					Name: "gated-klt",
```

Comment:
and I would move it into a common file 
```suggestion
					Name: "gated-keptn",
```